hibikase/Settings.h

```diff
@@ -14,14 +14,66 @@
 #pragma once
 
 #include <QByteArray>
+#include <QSettings>
 #include <QString>
 #include <QTextCodec>
 
+template<typename T>
+class Setting
+{
+public:
+    Setting(QString name, QString friendly_name, T default_value)
+        : name(name), friendly_name(friendly_name), default_value(default_value) {}
+
+    Setting(const Setting&) = delete;
+    Setting& operator=(const Setting&) = delete;
+
+    T Get()
+    {
+        std::unique_ptr<QSettings> settings = GetSettings();
+        return settings->value(name, default_value).template value<T>();
+    }
+
+    void Set(T value)
+    {
+        std::unique_ptr<QSettings> settings = GetSettings();
+        settings->setValue(name, value);
+        m_callback(value);
+    }
+
+    void Reset()
+    {
+        Set(default_value);
+    }
+
+    void SetCallback(std::function<void(T)> callback)
+    {
+        m_callback = callback;
+    }
+
+    const QString name;
+    const QString friendly_name;
+    const T default_value;
+
+private:
+    static std::unique_ptr<QSettings> GetSettings()
+    {
+        return std::make_unique<QSettings>(QSettings::Format::IniFormat, QSettings::Scope::UserScope, "MoonCat Karaoke", "Hibikase");
```

Comment:
If I'm reading the Qt source code correctly, `QSettings` reads from disk when constructed. Could we keep one static instance of `QSettings` somewhere?